api/src/main/java/marquez/db/models/ColumnLineageNodeData.java

```diff
@@ -42,6 +42,17 @@ public ColumnLineageNodeData(
     this.inputFields = inputFields;
   }
 
+  public ColumnLineageNodeData(InputFieldNodeData data) {
+    this.namespace = data.namespace;
+    this.dataset = data.dataset;
+    this.datasetVersion = data.datasetVersion;
+    this.field = data.field;
+    this.fieldType = "";
+    this.transformationDescription = data.transformationDescription;
+    this.transformationType = data.transformationType;
+    this.inputFields = ImmutableList.of();
+  }
```

Comment:
I'm a little confused by some of these fields like `fieldType` and `inputFields`. Would these ever be set here?

Reply:
The truth is that if I set `InputFieldNodeData` to implement `NodeData` interface jackson couldn't deduce what was type of `NodeData` was in the output node (the one that previously had `data` field null). For that reason I basically made a workaround to simply create `ColumnLineageNodeData` from `InputFieldNodeData`.

I'd need more guidance how to fix jackson deduction :|